### ISSUE TITLE
fingerprint verification via GUI

### ISSUE THREAD

**Issue by oodbur** [gkdr/lurch#48]
for the less nerdy people you should create something clickable to at least just show known fingerprints.
among other places, a config dialog is where people are used to find such information.
in the long run lurch should store key's verification states and support all the verification options that OTR offered, plus options from other OMEMO implementations like QR code.

**Comment by gkdr** [comment]
I agree that a GUI would make it easier for the usual user (and not just for fingerprint verfication), but that's out of scope of a _libpurple_ plugin (as it is right now it also works in Finch!).
There does exist to add an options dialog and I'm looking into that for displaying general information, however that won't for for conversation-specific data directly from the conversation window (AFAIK).
In any case, _Pidgin_-specific GUI extensions are planned for someday.

(The top priority for the near future will be implementing the new version of the OMEMO spec though.)

**Comment by traumschule** [comment]
+1

It is no option for me to remember `/lurch show fp conv `or to lookup in the README every time.

**Comment by wilhelmy** [comment]
It sounds like it would help if someone else would try to write frontend code for pidgin in the meantime?
Maybe I can squeeze that in at some point, but not in the next few months. I'm not yet familiar with the pidgin codebase but it sounds doable. I'll keep it in mind if nobody beats me to it.

**Comment by wilhelmy** [comment]
I think there is a chicken-and-egg problem with OMEMO adaption and having a readily-available user interface. Even as a technically capable user I was confused at first about how I use this. I think having an UI for fingerprint verification and known peer fingerprints would help me with daily use.

Thanks for writing this plugin. You're doing important work here.

**Comment by gkdr** [comment]
I'd certainly appreciate some help, though I should have more free time now (again).
I'm still not sure how to handle that it should be a libpurple plugin, but also have somewhat of a GUI. There could easily be a second plugin just for the GUI, but that's quite uncomfortable. I am not sure if I remember right but there might be the possibility to write Pidgin (i.e. GUI) code in the same plugin, but it will simply not be loaded in other libpurple programs.

**Comment by GigabyteProductions** [comment]
Perhaps they'd offer advice on the developers' mailing list?

See: https://pidgin.im/cgi-bin/mailman/listinfo/devel
